src/layout.rs

```diff
@@ -338,9 +338,12 @@ impl Layout {
     pub fn horizontal<I>(constraints: I) -> Layout
     where
         I: IntoIterator,
-        I::Item: AsRef<Constraint>,
+        I::Item: Into<Constraint>,
```

Comment:
I don't think we can change the new() impl to use Into instead of AsRef, as it would break too much code. Need to investigate the impact. These methods are not yeat released, so we can break code easily.

Making this accpet Into<Constraint> allows for:
```rust
Layout::horizontal([1,2,3,4])
```